storage/src/test/java/org/apache/kafka/tiered/storage/integration/BaseDeleteSegmentsTest.java

```diff
@@ -55,7 +56,7 @@ protected void writeTestSpecifications(TieredStorageTestBuilder builder) {
                 .expectSegmentToBeOffloaded(broker0, topicA, p0, 2, new KeyValueSpec("k2", "v2"))
                 .expectEarliestLocalOffsetInLogDirectory(topicA, p0, 3L)
                 .produceWithTimestamp(topicA, p0, new KeyValueSpec("k0", "v0"), new KeyValueSpec("k1", "v1"),
-                        new KeyValueSpec("k2", "v2"), new KeyValueSpec("k3", "v3", System.currentTimeMillis()))
+                        new KeyValueSpec("k2", "v2"), new KeyValueSpec("k3", "v3", System.currentTimeMillis() + TimeUnit.HOURS.toMillis(1)))
```

Comment:
Could we add a comment on why we need to use a future timestamp?

Reply:
When remote storage is enabled, we will roll a segment subject for deletion due to retention. This could eventually lead to the tiering and the deletion of the segment.

```
        // Roll the active segment when it breaches the configured retention policy. The rolled segment will be
        // eligible for deletion and gets removed in the next iteration.
        if (predicateResult && remoteLogEnabled() && nextSegmentOpt.isEmpty && segment.size > 0) {
          shouldRoll = true
        }
```

Reply:
Also, the error message in ConsumerAction is not very clear. It would be useful to include the expected count, the actual count and the OperationType.

`Number of FETCH_OFFSET_INDEX requests from broker 0 to the tier storage does not match the expected value for topic-partition topicA-0 ==> expected: <true> but was: <false>`